sanitycheck: Remove duplicate headers in CSV file and add named exceptions

nashif opened the pull request:
Dont merge test files with multiple header rows. 
Cleanup exceptions and logging.

galak commented on scripts/sanitycheck:
```diff
@@ -2646,7 +2646,8 @@ class TestSuite:
                 instance_list = []
                 for row in cr:
-                    if row["status"] in filter_status:
+                    # FIXME: ignore header row.
```
Can we explain the `row['arch'] == 'arch'` in the comment in a little more detail.

Something like:

```
# We might have header rows intermixed, detect them by the the 'arch' field being set to 'arch'
```

nashif committed:
sanitycheck: Add named exceptions

Add named exceptions instead of a generic one. Should help point to the
right issue when something goes wrong.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: when merging files, keep one header row

When merging files, make sure to remove header rows from each file and
just keep one header. This will stop us from parsing header rows as
tests.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

galak commented on scripts/ci/run_ci.sh:
```diff
@@ -251,7 +251,10 @@ if [ -n "$main_ci" ]; then
 	${sanitycheck} ${sanitycheck_options} --save-tests test_file_3.txt || exit 1
-	cat test_file_1.txt test_file_2.txt test_file_3.txt > test_file.txt
+
+	tail -n +2 test_file_2.txt > test_file_2_in.txt
```
can you add a comment here about skipping the header.

galak commented on scripts/sanitycheck:
```diff
@@ -2644,10 +2644,13 @@ class TestSuite:
             with open(file, "r") as fp:
                 cr = csv.DictReader(fp)
+                # Skip header
+                next(cr)
```
Is this really needed, I thought the DictReader will pull off the header?

nashif replied:
it is not, with the above change

nashif committed:
sanitycheck: remove confusing message when loading files

This message was intended for retries only, not every time we loaded a
file with a test manifest.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
